ontopy/ontology.py

```diff
@@ -984,13 +1001,12 @@ def save(
 
         elif write_catalog_file:
             write_catalog(
-                {self.get_version(as_iri=True): filename},
+                {self.get_version(as_iri=True): filepath},
                 output=catalog_file,
                 directory=dir,
                 append=append_catalog,
             )
-
-        if squash:
+        elif squash:
```

Comment:
This means that if you specify `write_catalog_file` you will not be able to squash. I agree that it is not a very sensible to write a catalogue file when squashing, but why prohibit it?

```suggestion
        if squash:
```

Reply:
The problem is then that when it does a recursive save it will then proceed through this next if/else part and actually write self again (in the wrong place). You can try.

Reply:
But we explicitly set `squash=False` when `save()` calls itself recursively.